src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,23 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private Engine engine;
+    private String colour;
+    private String model;
+    private int capacity;
+    private int maxSpeed;
 
+    public Car(Engine engine, String colour, String model, int capacity, int maxSpeed) {
+        this.engine = engine;
+        this.colour = colour;
+        this.model = model;
+        this.capacity = capacity;
+        this.maxSpeed = maxSpeed;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+        Car cloned = new Car(engine.clone(), colour, model, capacity, maxSpeed);
+        return cloned;
```

Comment:
don't create variable, just return the instance

Reply:
done